x-pack/test/functional/apps/endpoint/host_list.ts

```diff
@@ -67,21 +67,70 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
           'xxxx',
         ],
       ];
-      const tableData = await pageObjects.endpoint.getEndpointAppTableData('managementListTable');
+      const tableData = await pageObjects.endpoint.getEndpointAppTableData('hostListTable');
       expect(tableData).to.eql(expectedData);
     });
 
-    it('displays no items found', async () => {
+    it('display details flyout when the hostname is clicked on', async () => {
+      await (await testSubjects.find('hostnameCellLink')).click();
+      await testSubjects.existOrFail('hostDetailsUpperList');
+      await testSubjects.existOrFail('hostDetailsLowerList');
+    });
+
+    it('displays no items found when empty', async () => {
       // clear out the data and reload the page
       await esArchiver.unload('endpoint/metadata/api_feature');
-      await pageObjects.common.navigateToUrlWithBrowserHistory('endpoint', '/management');
+      await pageObjects.common.navigateToUrlWithBrowserHistory('endpoint', '/hosts');
       // get the table data and verify no entries appear
-      const tableData = await pageObjects.endpoint.getEndpointAppTableData('managementListTable');
+      const tableData = await pageObjects.endpoint.getEndpointAppTableData('hostListTable');
       expect(tableData[1][0]).to.equal('No items found');
       // reload the data so the other tests continue to pass
       await esArchiver.load('endpoint/metadata/api_feature');
     });
 
+    describe('has a url with a host id', () => {
+      before(async () => {
+        await pageObjects.common.navigateToUrlWithBrowserHistory(
+          'endpoint',
+          '/hosts',
+          'selected_host=cbe80003-6964-4e0f-aba1-f94c32b44e95'
+        );
+      });
+
+      it('shows a flyout', async () => {
+        await testSubjects.existOrFail('hostDetailsFlyout');
+      });
+
+      it('displays details row headers', async () => {
+        const expectedData = [
+          'OS',
+          'Last Seen',
+          'Alerts',
+          'Policy',
+          'Policy Status',
+          'IP Address',
+          'Hostname',
+          'Sensor Version',
+        ];
+        const keys = await pageObjects.endpoint.hostFlyoutDescriptionKeys('hostDetailsFlyout');
+        expect(keys).to.eql(expectedData);
+      });
+
+      it('displays details row descriptions', async () => {
+        const values = await pageObjects.endpoint.hostFlyoutDescriptionValues('hostDetailsFlyout');
+
+        expect(values).to.eql([
+          'Windows Server 2012',
+          '',
+          '0',
+          'C2A9093E-E289-4C0A-AA44-8C32A414FA7A',
+          'active',
+          '10.48.181.22210.116.62.6210.102.83.30',
```

Comment:
can we also enhance the way this list of IP addresses is displayed. maybe work for next PR

Reply:
when you say enhance do you mean in terms of how it's displayed here in the test code or did you mean in the actual ui?

Reply:
i just mean there's no separator between the IPs here and i think in the UI also